osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/services/RedisPermitService.java

```diff
@@ -65,22 +70,40 @@ public boolean createPermit(
 
           log.debug(
               "Trying to register a permit for request[{}] with permit key {}. (max-concurrent-requests: {}, number-of-registered-permits: {})",
-              requestId,
+              permitRequest.getRequestId(),
               permitKey.key(),
               maxConcurrentRequests,
               numberOfRegisteredPermits);
 
-          if (maxConcurrentRequests < 0 || numberOfRegisteredPermits < maxConcurrentRequests) {
-            granted =
-                permits.add(
-                    Instant.now().toEpochMilli(), new Permit(networkSegment, clientId, requestId));
+          // get waiting room for key
+          final RScoredSortedSet<PermitRequest> lobby =
+              this.redisson.getScoredSortedSet(permitKey.lobby());
+
+          // if prio = default and waiting room size > 0 => EXIT (granted = false)
+          if (highPrio || lobby.isEmpty()) {
+
+            if (maxConcurrentRequests < 0 || numberOfRegisteredPermits < maxConcurrentRequests) {
+              granted =
+                  permits.add(
+                      Instant.now().toEpochMilli(), new Permit(networkSegment, permitRequest));
+            }
+
+            if (highPrio) {
+              if (granted) {
+                // if granted = true and prio = high and in waiting room => remove from waiting room
```

Comment:
Where is the _in waiting room_ check in code?

Reply:
RScoredSortedSet#remove only removes if present and does not require an explicit check for existence.